src/app/tabs/compileTab/compilerContainer.js

```diff
@@ -240,7 +248,9 @@ class CompilerContainer {
 
   onchangeLanguage (event) {
     this.compileTabLogic.setLanguage(event.target.value)
-    this.compile()
```

Comment:
all the other `on*` method, just trigger a new compilation. Might be more consistent to to keep the same behavior here

Reply:
but there was a issue here. build shouldn't be triggered if there is no autocompile an selected file

Reply:
or I can put the check in the compile() function

Reply:
> but there was a issue here. build shouldn't be triggered if there is no autocompile an selected file

This should done in another PR and all the `on*` function should be updated.

Reply:
ok will create a new PR